tools/autograd/derivatives.yaml

```diff
@@ -581,6 +602,9 @@
   self: grad
   src: grad.gather(dim, index)
 
+- name: select(Tensor self, int64_t dim, int64_t index)
```

Comment:
I added custom backward for `select` and `slice` (and consequently `narrow`) because they should be more efficient than `as_strided_backward` in most cases.